### DISCUSSION TITLE
How can I temporarily disable all workflows in the repository settings?

### DISCUSSION THREAD

**Discussion by masakaya** [community/community#58971] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, I have a question. How can I temporarily disable all workflows in the repository settings?

I have cloned the repository in the following way
This is because I wanted to create a new product while keeping the past commit history.

1. `git clone --bare <OLD-REPOSITORY>` 
2. `git push --mirror <NEW-REPOSITORY>`

At this time,
`OLD-REPOSITORY` workflows were executed at the timing of Push to `NEW-REPOSITORY`.

To prevent such an incident, we would like to temporarily disable all workflows.

Thank you in advance for your help.

**Comment by airtower-luna** [comment, marked as answer]
Check the repository settings under Actions > General. You can disable Actions there.

**Comment by kaddkaka** [comment]
Where can I find that setting?

**Reply by masakaya** [reply]
@kaddkaka 

refer to the following image :)

![image](https://github.com/community/community/assets/1237574/99168c90-5934-48ab-b2f5-f3eda75842d9)

**Comment by gurjeet** [comment]
I disabled workflows by selecting 'Disable Actions' and clicking 'Save' button below the list. My `git push --mirror` are still being rejected with the following error.

```
 ! [remote rejected] main -> main (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/ci.yaml` without `workflow` scope)
 ! [remote rejected] v0.1.0 -> v0.1.0 (refusing to allow a Personal Access Token to create or update workflow `.github/workflows/ci.yaml` without `workflow` scope)
```

**Reply by airtower-luna** [reply]
That has nothing to do with Actions being enabled or disabled. Your token needs the `workflow` scope to push changes that change workflow files, no matter if the workflows will be executed after the push.

**Reply by majpue** [reply]
Same situation brought me here while mirroring a public -> private repo.
Using `gh` cli, I just had to [add the scope](https://cli.github.com/manual/gh_auth_refresh):

`gh auth refresh --scopes workflow`

**Reply by gurjeet** [reply]
> That has nothing to do with Actions being enabled or disabled. Your token needs the `workflow` scope to push changes that change workflow files, no matter if the workflows will be executed after the push.

@airtower-luna Thanks for the response! I enabled the `workflow` permission for my token, and that allowed me to push the repo that contained workflow.

But the thing I'm not comfortable with is that that Github initiates the workflows/Actions right after the first time I push the repository. I am mirroring a repository merely for posterity, so I want to disable any workflow executions. Do you have any recommendations on how to do that via the REST API? The [repository creation API endpoint documentation][] does not have any mention of Actions/Workflows.

[repository creation API endpoint documentation]: https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#create-an-organization-repository

Just for my future reference, I faced this problem when I was trying to create a mirror (not a fork) of the [dufs][] repository, using Github's REST API via curl.

[dufs]: https://github.com/sigoden/dufs

**Comment by MekongDelta-mind** [comment]
1. Go to  `Settings` tab of your repo.
2. Then select `Actions` on the left side panel drop down.
3. Then click on `General` .
4. You will have `Actions Permissions` on the right side panel. Click on `Disable Actions` and save it.
![image](https://github.com/user-attachments/assets/3f15140c-5617-4358-9cd8-e97c11f1ec8e)
